cpp/include/cudf/strings/detail/gather.cuh

```diff
@@ -109,28 +110,25 @@ __global__ void gather_chars_fn_string_parallel(StringIterator strings_begin,
     // between `[out_start_aligned, out_end_aligned)` will be copied using uint4.
     // `out_start + 4` and `out_end - 4` are used instead of `out_start` and `out_end` to avoid
     // `load_uint4` reading beyond string boundaries.
-    int32_t out_start_aligned =
+    int64_t const out_start_aligned =
```

Comment:
Shall we use an alias to avoid using raw `int64_t` all over the place? e.g.
```cpp
  using offset_type = typename cudf::detail::input_offsetalator::value_type;
  offset_type const out_start_aligned = ...
```

Reply:
I'd rather not do this at this time. Since offset-types can be either int32 or int64, I do not want to use `offset_type` as a name. I'd prefer using `int64` for the time being.

Reply:
Explicit type is preferred based on offline discussions.